pkg/server/server.go

```diff
@@ -175,15 +182,16 @@ func (s *ProxyServer) getFrontend(agentID string, connID int64) (*ProxyClientCon
 
 // NewProxyServer creates a new ProxyServer instance
 func NewProxyServer(serverID string, serverCount int, agentAuthenticationOptions *AgentTokenAuthenticationOptions) *ProxyServer {
-	p := &ProxyServer{
+	bm := NewDefaultBackendManager()
+	return &ProxyServer{
 		frontends:                  make(map[string](map[int64]*ProxyClientConnection)),
 		PendingDial:                NewPendingDialManager(),
 		serverID:                   serverID,
 		serverCount:                serverCount,
-		BackendManager:             NewDefaultBackendManager(),
+		BackendManager:             bm,
 		AgentAuthenticationOptions: agentAuthenticationOptions,
+		Readiness:                  bm,
```

Comment:
Ideally we should have something (ReadinessManager) which aggregates a list of Readiness checks. Any chance we can add the infra now so its easy to additional readiness checks as we find the need?

Reply:
I did define the ReadinessManager as an interface. Currently I make the backendManager implements the interface. We can compose a dedicated readinessManager with more checks in the future. Is that ok?

Reply:
Sure